protodroid/src/main/java/id/lukasdylan/grpc/protodroid/internal/repository/InternalProtodroidRepository.kt

```diff
@@ -4,17 +4,71 @@ import id.lukasdylan.grpc.protodroid.internal.database.ProtodroidDao
 import id.lukasdylan.grpc.protodroid.internal.database.ProtodroidDataEntity
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.withContext
 
 internal interface InternalProtodroidRepository {
     suspend fun fetchAllData(): Flow<List<ProtodroidDataEntity>>
     suspend fun fetchSingleData(dataId: Long): Flow<ProtodroidDataEntity>
     suspend fun deleteAllData()
+    suspend fun getFilteredData(): Flow<List<ProtodroidDataEntity>>
```

Comment:
Better make this function had a parameter list of filter, instead of put on a variable